internal/olm/operator/internal/registry_pod.go

```diff
@@ -297,25 +289,11 @@ func (rp *RegistryPod) getContainerCmd() (string, error) {
 	return out.String(), nil
 }
 
-// GetLogs gets the logs for the registry pod
-// and throws error if failed to get pod logs
-func (rp *RegistryPod) GetLogs(ctx context.Context) (string, error) {
+// GetPod returns the registry pod which is of type corev1.Pod
```

Comment:
When will the caller ever need to get the pod itself?

Reply:
Will use it in IndexImageCatalogCreator to be able to set the pod as the owner to cs, as shown below:

`	if err = registryPod.Create(ctx); err != nil {
		return nil, fmt.Errorf("error in creating registry pod")
	}

	pod, err := registryPod.GetPod(ctx)
	if err != nil {
		return nil, fmt.Errorf("error in creating registry pod")
	}

	if err := controllerutil.SetOwnerReference(cs, pod, c.cfg.Scheme); err != nil {
		return nil, fmt.Errorf("unable to set registry pod owner reference: %v", err)
	}`

Reply:
In the packagemanifests logic, we also have to do this for registry objects. What if we had a command that sets owner references on arbitrary objects like

```go
func (rp *RegistryPod) SetOwnerReferences(owner controllerutil.Object, sch *runtime.Scheme) error {
	return controllerutil.SetOwnerReferences(owner, rp.pod, sch)
}
```

Once we have an interface to wrap `RegistryPod` and `RegistryResources`, this can be a method that both would implement.

Reply:
```
       if err = registryPod.Create(ctx); err != nil {
		return nil, fmt.Errorf("error in creating registry pod")
	}

	pod, err := registryPod.GetPod(ctx)
	if err != nil {
		return nil, fmt.Errorf("error in creating registry pod")
	}

	if err := controllerutil.SetOwnerReference(cs, pod, c.cfg.Scheme); err != nil {
		return nil, fmt.Errorf("unable to set registry pod owner reference: %v", err)
	}

Reply:
@estroz meant to paste this example in the earlier comment of mine :) 

The example you sent - are you suggesting that `SetOwnerReferences` will be a method on an interface and any type that wants to set the owner will call it (e.x. RegistryPod). I guess that will work once we wrap `RegistryPod` and `RegistryResources` as you suggested.

Reply:
In the current design we have for `run bundle`, we would want to create registry pod and catalog source and assign ownership over here 
https://github.com/operator-framework/operator-sdk/pull/3636/files#diff-73dbfb52b8b7117d3433faafe1cd8192R42

I think this will be OK for now. And once we have the logic to wrap` RegistryPod` and `RegistryResources`, this can be a method that both would implement.

Reply:
Yeah. In internal:
```go
type RegistryManager interface {
	Create(ctx context.Context, namespace string) error
	Delete(ctx context.Context, namespace string) error
	SetOwnerReferences(owner controllerutil.Object, sch *runtime.Scheme) error
}

var _ RegistryManager = &RegistryResources{}
var _ RegistryManager = &RegistryPod{}
```

in the caller:

```go
func (i PackageManifestsInstall) Run() error {
	catsrc, _ := i.createCatalogSource()
	m, _ := internal.NewRegistryResources(i.pkg, i.bundles, i.version)
	_ = m.SetOwnerReferences(catsrc, i.scheme)
	_ = m.Create(ctx, i.namespace)
	...
}

func (i BundleInstall) Run() error {
	catsrc, _ := i.createCatalogSource()
	m, _ := internal.NewRegistryPod(i.bundle)
	_ = m.SetOwnerReferences(catsrc, i.scheme)
	_ = m.Create(ctx, i.namespace)
	...
}
```

Reply:
Sure, that makes sense. Removed GetPod function in this PR